js/settingsHelpers.js

```diff
@@ -0,0 +1,408 @@
+  isFormdisplayed = !isFormdisplayed;
+  elementArray.forEach(element => {
+    if (element !== null) {
+      element.style.display === 'none' ? (element.style.display = 'block') : (element.style.display = 'none');
+    }
+  });
+
+  return isFormdisplayed;
+};
+
+export const suffixList = { 612166858: 0, 255907182: 1, 226924545: 2, 270793412: 3, 959021713: 4, 643664527: 5, 537892528: 6 }
+
+export const suffixToTextMap = new Map([
+  [612166858, 'Jr.'],
+  [255907182, 'Sr.'],
+  [226924545, 'I'],
+  [270793412, 'II'],
+  [959021713, 'III'],
+  [643664527, '2nd'],
+  [537892528, '3rd'],
+])
+
+export const togglePendingVerificationMessage = userData => {
+  const isUserSubmitted = userData[cId.userProfileSubmittedAutogen] === cId.yes;
+  const isUserVerified = userData[cId.verified] === cId.yes;
+  const verificationMessage = document.getElementById('pendingVerification');
+  isUserSubmitted && !isUserVerified ? (verificationMessage.style.display = 'block') : (verificationMessage.style.display = 'none');
+};
+
+export const hideSuccessMessage = successMessageElement => {
+  if (successMessageElement) {
+    successMessageElement.style.display = 'none';
+  }
+  return null;
+};
+
+export const handleOptionalFieldVisibility = (value, text, element, matcher, type) => {
+  if (value) {
+    if (type === 'text') {
+      document.getElementById(text).textContent = value;
+    } else if (type === 'suffix') {
+      document.getElementById(text).textContent = suffixToTextMap.get(parseInt(value));
+    } else if (type === 'phone') {
+      document.getElementById(text).textContent = `${value.substring(0, 3)} - ${value.substring(3, 6)} - ${value.substring(6, 10)}`;
+    } else {
+      console.log('ERROR: bad type expression in handleOptionalFieldVisibility');
+    }
+    element.style.display = matcher.style.display;
+  } else {
+    element.style.display = 'none';
+  }
+};
+
+export const showAndPushElementToArrayIfExists = (value, element, array) => {
+  if (value) {
+    element.style.display = 'block';
+    array.push(element);
+  }
+};
+
+export const handleContactInformationRadioButtonPresets = (canWeVoicemailMobile, canWeText, canWeVoicemailHome) => {
+  document.getElementById('mobileVoicemailPermissionYesRadio').checked = canWeVoicemailMobile;
+  document.getElementById('mobileVoicemailPermissionNoRadio').checked = !canWeVoicemailMobile;
+  document.getElementById('textPermissionYesRadio').checked = canWeText;
+  document.getElementById('textPermissionNoRadio').checked = !canWeText;
+  document.getElementById('homeVoicemailPermissionYesRadio').checked = canWeVoicemailHome;
+  document.getElementById('homeVoicemailPermissionNoRadio').checked = !canWeVoicemailHome;
+};
+
+export const FormTypes = {
+  NAME: 'nameForm',
+  CONTACT: 'contactForm',
+  MAILING: 'mailingForm',
+  EMAIL: 'emailForm',
+};
+
+export const validEmailFormat = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+
+export const updatePhoneNumberInputFocus = () => {
```

Comment:
Should we create common function like

const focusEvent = (id) => {
const element = document.getElementById(id);
  element.addEventListener('keyup', () => {
    if (element.value.trim().length === 3) {
      element.nextElementSibling.nextElementSibling.focus();
    }
  });
}

Reply:
Great suggestion. It's only called on update contact information, so I'll just iterate the array.

`
export const updatePhoneNumberInputFocus = () => {
  const phoneElementIds = ['mobilePhoneNumber1', 'mobilePhoneNumber2', 'homePhoneNumber1', 'homePhoneNumber2', 'otherPhoneNumber1', 'otherPhoneNumber2'];

  const initFocusHandler = elementId => {
    const element = document.getElementById(elementId);
    element.addEventListener('keyup', () => {
      if (element.value.trim().length === 3) {
        element.nextElementSibling.nextElementSibling.focus();
      }
    });
  };

  phoneElementIds.forEach(initFocusHandler);
};
`